doc/developers-guide/accumulators.md

```diff
@@ -30,132 +30,111 @@ a) performance - if we need to do something involving inspecting a stanza or mor
-When the last stanza in the whole process is leaving the system we have a complete record of what we have sent in the meantime.
-
-An 'exit point' is where we call a function sending something out of the system and, if all goes well, returning the original accumulator possibly with an additional track record of what's just been done.
-
-See the diagram to examine this concept.
-Note that Acc1 going to the routing has the same `ref` as the one returning to the next function:
-
-![You should see an image here; if you don't, use plantuml to generate it from accum_path.uml](accum_path.png)
-
-
-An 'exit point' function should be called with an acc AND a stanza to be sent out. 
-Sometimes the stanza is the original one (which is stored in the accumulator as `element`), sometimes it's not.
-The 'exit point' should return an accumulator.
-
-### `ejabberd_c2s:send_element/3`
+## IQs and accumulators
 
-Here we send a stanza to an XMPP client.
-It is often called directly by a c2s process when it needs to respond to its "owner".
+`mongoose_iq` module exposes dedicated API for accessing IQ-related accumulator fields. These are:
 
-### `ejabberd_router:route/4`
+* `info(Acc)` - Returns #iq{} record produced from a stanza stored in the accumulator. May be `invalid` or `not_iq` if the stanza is not a valid IQ.
+* `xmlns(Acc)` - Returns XMLNS of the first subelement inside an IQ. In most cases it is a namespace of `<query/>` subelement. May be `undefined`.
+* `command(Acc)` - Returns the name of a first subelement inside an IQ. May be `undefined`.
 
-This function is called when a stanza is ready to be sent.
-It makes an attempt to route it out via the appropriate channel - either locally to the same or another c2s process, or somewhere else via s2s.
-The stanza can be sent or dropped, or it may error out - whatever it does this function returns the original accumulator, optionally with the track record of what it tried to send, how and to what effect (see the commented out code in `mongoose_acc:record_sending/4`).
-
-Under the hood the routing modules work by passing an original accumulator and a stanza which is to be sent, which may be the original
-xml element, same element with modifications, or a broadcast.
-
-Eventually there are three things that may happen:
-* the accumulator is sent to another c2s process, pubsub node or muc room - in this case we produce another fresh accumulator out of the data to be sent, stripping it of all caches and the likes but preserving original ref and timestamp and those attrs that should be carried on (`mongoose_acc:strip/2`)
-* stanza is stored as offline
-* stanza is sent out to another XMPP server via `ejabberd_s2s`.
+These functions ensure that cached information matches accumulator's stanza, so all of them return a tuple with a possibly updated acc as a second element.
```

Comment:
These functions ensure that cached information matches the accumulator's stanza, so all of them return a tuple with a possibly updated acc as a second element.